core/directory.go

```diff
@@ -267,7 +267,31 @@ func (dir *Directory) WithDirectory(ctx context.Context, subdir string, src *Dir
 	return destPayload.ToDirectory()
 }
 
-func (dir *Directory) WithCopiedFile(ctx context.Context, subdir string, src *File) (*Directory, error) {
+func (dir *Directory) WithNewDirectory(ctx context.Context, gw bkgw.Client, dest string) (*Directory, error) {
+	payload, err := dir.ID.Decode()
+	if err != nil {
+		return nil, err
+	}
+
+	// be sure to create the file under the working directory
+	dest = path.Join(payload.Dir, dest)
```

Comment:
Does this handle `dest` being something like `../../../foo` when `payload.Dir` is not `""`? Seems like it might succeed but then could just result in no visible change to the actual directory. Not the end of the world but for consistency maybe this should be an error case.

Reply:
Added an error check + test!